wadsrc/static/zscript/engine/base.zs

```diff
@@ -638,7 +677,8 @@ struct StringStruct native
 {
 	native static vararg String Format(String fmt, ...);
 	native vararg void AppendFormat(String fmt, ...);
-
+	// native int Length();  // Intrinsic
+	// bool operator~==(String other)  // Case-insensitive equality comparison
```

Comment:
I would leave `StringStruct` alone, and create `class String` with all its members.
Alternatively, you should mention `StringStruct` relation to internal `string` type to avoid copy-pasting of declarations.

Also, `operator.()` needs to added to these functions.